tic_tac_toe/game.py

```diff
@@ -87,18 +135,54 @@ def move(game, player, position):
     checks before the actual movement is done.
     After registering the movement it must check if the game is over.
     """
-    pass
-
+    # check input
+    if (_board_is_full(game['board'])) or (game['winner'] != None):
+        raise InvalidMovement("Game is over.")
+    if game['next_turn'] != player:
+        if game['next_turn'] == 'O':
+            raise InvalidMovement('"O" moves next.')
+        else:
+            raise InvalidMovement('"X" moves next.')
+    if  not _position_is_valid(position):
+        raise InvalidMovement('Position out of range.')
+    if not _position_is_empty_in_board(position, game['board']):
+        raise InvalidMovement('Position already taken.')
+        
+    # make move, toggle next player
+    game['board'][position[0]][position[1]] = player
+    if player == game['player1']:
+        game['next_turn'] = game['player2']
+    elif player == game['player2']:
+        game['next_turn'] = game['player1']
+    
+    
+    # check game endings
+    game['winner'] = get_winner(game)
+    if game['winner'] == game['player1']:
+        raise GameOver('"X" wins!')
+    elif game['winner'] == game['player2']:
+        raise GameOver('"O" wins!')
+    elif _board_is_full(game['board']) and game['winner'] == None:  
```

Comment:
Don't need to check if there is a game winner at this point as if there was your previous if statements would catch it.